packages/server/src/router.ts

```diff
@@ -283,10 +280,8 @@ export class Router<
   ): Router<
     TInputContext,
     TContext,
-    flatten<
-      TQueries,
-      Record<TPath, inferProcedureFromOptions<TInputContext, typeof procedure>>
-    >,
+    TQueries &
+      Record<TPath, inferProcedureFromOptions<TInputContext, typeof procedure>>,
```

Comment:
I wonder if this breaks routers which redefine queries with a duplicate path, since flatten made the lhs "win"? Probably a good thing if so since it would almost always be a mistake, maybe?

Reply:
Duplicate queries are already handled and throws an error

Reply:
https://github.com/trpc/trpc/blob/2c1341f182df825509bcb7bd546fd724fd1f58a6/packages/server/src/router.ts#L486

Reply:
Let me see though

Reply:
Duplicate queries should already throw an error!